src/main/java/com/miraisolutions/xlconnect/Workbook.java

```diff
@@ -320,56 +331,63 @@ public void renameSheet(String name, String newName) {
     }
 
     public void cloneSheet(int index, String newName) {
-        cloneSheet(workbook.getSheetName(index), newName);
+        Sheet sheet = workbook.cloneSheet(index);
+        String originalName = workbook.getSheetName(index);
+        workbook.setSheetName(workbook.getSheetIndex(sheet), newName);
+        // Copy names (named ranges) that are scoped to the original sheet, adapting the scope.
+        List<Name> originalNamedRanges = workbook.getAllNames().stream().filter(name -> name.getSheetIndex() == index).collect(Collectors.toList());
+        // we have to collect the original names *then* add the new names, otherwise we get concurrency issues creating names while iterating over them
+        originalNamedRanges.forEach(namedRange ->
+                createName(namedRange.getNameName(), newName, namedRange.getRefersToFormula(), false));
     }
 
     public void cloneSheet(String name, String newName) {
-        Sheet sheet = workbook.cloneSheet(workbook.getSheetIndex(name));
-        workbook.setSheetName(workbook.getSheetIndex(sheet), newName);
+        cloneSheet(workbook.getSheetIndex(name), newName);
     }
-    
-    public void createName(String name, String formula, boolean overwrite) {
-        if(existsName(name)) {
+
+    public void createName(String name,  String worksheetScope, String formula, boolean overwrite) {
+        if(existsName(name, worksheetScope)) {
             if(overwrite) {
                 // Name already exists but we overwrite --> remove
-                removeName(name);
+                removeName(name, worksheetScope);
             } else {
                 // Name already exists but we don't want to overwrite --> error
-                throw new IllegalArgumentException("Specified name '" + name + "' already exists!");
+                throw new IllegalArgumentException("Specified name '" + name + "' already exists in " + worksheetScope);
```

Comment:
We should perhaps handle explicitly cases where `worksheetScope` is `null` in the error message